pkg/agent/util/net_linux.go

```diff
@@ -376,3 +378,103 @@ func renameHostInterface(oriName string, newName string) error {
 	}
 	return nil
 }
+
+// PrepareHostInterfaceConnection prepares uplink connection to the OVS bridge client by:
+// 1. Renaming the uplink (a bridged suffix will be added to it).
+// 2. Creating an internal port (original name of the uplink will be used here).
+// 3. Moving IPs of uplink to this new link/internal-port.
+// 4. Moving routes of uplink to the new link/internal-port.
+// and returns the bridged name, true if it already exists, and error.
+func PrepareHostInterfaceConnection(bridge ovsconfig.OVSBridgeClient, linkConfig *config.AdapterNetConfig, externalIDs map[string]interface{}) (string, bool, error) {
+	bridgedName := GenerateUplinkInterfaceName(linkConfig.Name)
+	// If the uplink port already exists, just return.
+	if uplinkOFPort, err := bridge.GetOFPort(bridgedName, false); err == nil {
+		klog.InfoS("Uplink already exists, skip the configuration", "uplink", bridgedName, "port", uplinkOFPort)
+		return "", true, nil
+	}
+
+	if err := RenameInterface(linkConfig.Name, bridgedName); err != nil {
+		return "", false, err
+	}
+	if _, err := bridge.CreateInternalPort(linkConfig.Name, int32(linkConfig.OFPort), linkConfig.MAC.String(), externalIDs); err != nil {
```

Comment:
Here `linkConfig.OFPort` is supposed to be prepared for the uplink not the host interface (internal port), it seems a misuse?

Reply:
Good catch.

Reply:
I'm passing the following config to `PrepareHostInterfaceConnection` from `ConnectUplinkToOVSBridge` now.
```go
&config.AdapterNetConfig{
		Name:   uplinkNetConfig.Name,
		OFPort: i.nodeConfig.HostInterfaceOFPort,
		Index:  uplinkNetConfig.Index,
		MAC:    uplinkNetConfig.MAC,
		IPs:    uplinkNetConfig.IPs,
		Routes: uplinkNetConfig.Routes,
	}
```